packages/preset-create-react-app/src/index.ts

```diff
@@ -1,20 +1,30 @@
 import { join, relative, resolve } from 'path';
 import { Configuration } from 'webpack'; // eslint-disable-line import/no-extraneous-dependencies
 import { logger } from '@storybook/node-logger';
+import PnpWebpackPlugin from 'pnp-webpack-plugin';
 import { mergePlugins } from './helpers/mergePlugins';
-import { getReactScriptsPath } from './helpers/getReactScriptsPath';
+import {
+  getReactScriptsPath,
+  getReactScriptsPathWithYarnPnp,
+} from './helpers/getReactScriptsPath';
 import { processCraConfig } from './helpers/processCraConfig';
 import { checkPresets } from './helpers/checkPresets';
 import { getModulePath } from './helpers/getModulePath';
 import { Options } from './options';
 
 const CWD = process.cwd();
-const REACT_SCRIPTS_PATH = getReactScriptsPath();
+// When operating under PnP environments, this value will be set to a number
+// indicating the version of the PnP standard, see: https://yarnpkg.com/advanced/pnpapi#processversionspnp
+const IS_USING_YARN_PNP = typeof process.versions.pnp !== 'undefined';
+const REACT_SCRIPTS_PATH = IS_USING_YARN_PNP
+  ? getReactScriptsPathWithYarnPnp()
+  : getReactScriptsPath();
 const OPTION_SCRIPTS_PACKAGE = 'scriptsPackageName';
 
 // This loader is shared by both the `managerWebpack` and `webpack` functions.
 const resolveLoader = {
   modules: ['node_modules', join(REACT_SCRIPTS_PATH, 'node_modules')],
+  plugins: [PnpWebpackPlugin.moduleLoader(module)],
```

Comment:
@gaetanmaisse I'm assuming this is a no-op when the user is not using Yarn PNP?

Reply:
✅ Yep exactly